cmd/xl-v1-utils.go

```diff
@@ -99,19 +101,89 @@ func hashOrder(key string, cardinality int) []int {
 	return nums
 }
 
+// Constructs XLMetaV1 using `gjson` lib to retrieve each field.
+func xlMetaV1UnmarshalJSON(xlMetaBuf []byte) (xlMetaV1, error) {
+
+	xlMeta := xlMetaV1{}
+	// obtain version.
+	xlMeta.Version = gjson.Get(string(xlMetaBuf), "version").String()
+	// obtain format.
+	xlMeta.Format = gjson.Get(string(xlMetaBuf), "format").String()
+	// fetching modTime.
+	modTime, err := time.Parse(time.RFC3339, gjson.Get(string(xlMetaBuf), "stat.modTime").String())
+	if err != nil {
+		return xlMetaV1{}, err
+	}
+	xlMeta.Stat.ModTime = modTime
+	// obtain Stat.Size .
+	xlMeta.Stat.Size = gjson.Get(string(xlMetaBuf), "stat.size").Int()
+	// parse the xlV1Meta.Erasure fields.
+	xlMeta.Erasure = erasureInfo{}
+	erasureResult := gjson.Get(string(xlMetaBuf), "erasure")
+	// parse the xlV1Meta.Erasure.Distribution.
+	disResult := erasureResult.Get("distribution").Array()
+
+	distribution := make([]int, len(disResult))
+	for i, dis := range disResult {
+		distribution[i] = int(dis.Int())
+	}
+	xlMeta.Erasure.Distribution = distribution
+
+	xlMeta.Erasure.Algorithm = erasureResult.Get("algorithm").String()
+	xlMeta.Erasure.DataBlocks = int(erasureResult.Get("data").Int())
+	xlMeta.Erasure.ParityBlocks = int(erasureResult.Get("parity").Int())
+	xlMeta.Erasure.BlockSize = erasureResult.Get("blockSize").Int()
+	xlMeta.Erasure.Index = int(erasureResult.Get("index").Int())
+	// Pare xlMetaV1.Erasure.Checksum array.
```

Comment:
Each of these individual blocks can be made into their own functions which can be re-purposed elsewhere.